sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -225,6 +237,9 @@ export class EventHubReceiver extends LinkEntity {
       try {
         this._onMessage!(receivedEventData);
       } catch (err) {
+        if (this._aborter) {
+          this._aborter.removeEventListener("abort", this._onAbort);
+        }
```

Comment:
Why are we removing the listener here? On error we dont stop the receiver, we just call the user's error callback. We should remove the listener only when the receive operation completes or is cancelled